### PR TITLE
HTTP serve SPA applications

### LINKED ISSUE #295: HTTP serve SPA applications
## Is your feature request related to a problem? Please describe

We would like Patron to serve the SPA files in order to host the SPA from within the micro service.

## Is this a bug? Please provide steps to reproduce, a failing test etc.

Not a bug.

## Describe the solution

The solution would be approximately the following:
- Add an option to the HTTP compenent in order to support servicng files
- The patron service, which always start a HTTP component, should also have the option to set this up.

### PR DESCRIPTION
**Pull request by kasperbe**
## Which problem is this PR solving?

Closes #295 

## Short description of the changes

Adds fileserver support for Single page applications.

### CODE REVIEW

File: component/http/route.go [hunk 1]
```diff
@@ -194,6 +194,61 @@ func (rb *RouteBuilder) Build() (Route, error) {
...
+		_, err := os.Stat(fallbackPath)
+		if os.IsNotExist(err) {
+			ee = append(ee, errors.New("fallback file doesn't exist"))
+		} else if err != nil {
+			ee = append(ee, fmt.Errorf("error while checking fallback file: %w", err))
+		}
+	}
+
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		params := httprouter.ParamsFromContext(r.Context())
```
Comment (mantzas) [inline component/http/route.go]: Is there a way to avoid httprouter bleeding in here?

**Review by mantzas — CHANGES_REQUESTED** [review]

File: component/http/route.go (hunk 1 above)
Comment (kasperbe) [reply component/http/route.go]: I am not sure if there's an existing way to do it, we have to reach into it for the `httprouter.ParamsFromContext` which gets the wildcard routing from the underlying router. 

I suppose we can wrap this ourselves in the router abstraction of patron, should I go ahead and fix that?

Comment (tpaschalis) [reply component/http/route.go]: We have an exported `http.ExtractParams` helper to expose parameters in a `map[string]string`. 

It was initially needed for better handling of RawRoutes parameters; could it also serve us here?

Comment (kasperbe) [reply component/http/route.go]: Let me check it out and see what I can do!

Comment (kasperbe) [reply component/http/route.go]: Yep, that did the trick! 💪

**Review by mantzas — DISMISSED** [review]

**Comment by tpaschalis** [comment]
Nice work @kasperbe, LGTM! Thanks for taking the time to contribute to Patron and improve its functionality for each and every user!

Just one small note. How do you feel about mentioning this new functionality on [the relevant documentation](https://github.com/beatlabs/patron/blob/master/docs/components/HTTP.md) page?   
I was thinking like a sentence or two along the lines of the `Raw RouteBuilder Constructor` section. ^^

**Review by tpaschalis — APPROVED** [review]
LGTM! Once again, thanks you for the initiative! ✈️

**Review by mantzas — APPROVED** [review]

**Pull request merged**